Battle.java

```diff
@@ -16,27 +15,30 @@ public static void battle(Entity player1, Entity player2) {
         printMessage("The Battle begins.");
         printMessage(player1.getName() + " and " + player2.getName() + " clash!");
 
-        while (player1.isAlive() && player2.isAlive()) {
+        player1.setActiveCooldown(0);
+        player2.setActiveCooldown(0);
 
-            printSeparator('*', 20);
-            printMessage(i + ". round:");
-
-            player1.attack(player2);
-            healthCheck(player1, player2);
-
-            if (player2.isAlive()) {
-                printSeparator('*', 20);
+        do {
+            double minCooldown = Math.min(player1.getActiveCooldown(), player2.getActiveCooldown());
+            player1.setActiveCooldown(player1.getActiveCooldown() - minCooldown);
+            player2.setActiveCooldown(player2.getActiveCooldown() - minCooldown);
+            if (player1.getActiveCooldown() == 0) {
+                player1.attack(player2);
+            }
+            if (player2.getActiveCooldown() == 0) {
                 player2.attack(player1);
-                healthCheck(player1, player2);
             }
-            i++;
-        }
+        } while(player1.isAlive() && player2.isAlive());
```

Comment:
Letesztelted ezt? Nekem ugy tunik, hogy az egyik lemegy 0-ra, a min utana 0 lesz, es ezert csak az ut folyamatosan, akinek a cooldownja kezdetben kisebb.

Reply:
Sajnos nem tudtam, mert Adolf hotfixére vártam, szerintem viszont jó. A cooldownt reseteljük az attack függvényben, az meg hogy támadnak azok meg függetlenek egymástól. DE elismerem hülye voltam, hogy tesztelés nélkül re-requestet kértem. Ránézek mégegyszer.

Reply:
Oh, hat igen, normal esetben csak lefordulo kod van git-en :-) De semmi baj nincs, ezekbol tanulunk. A tegnapi ora utan mar latod, hogy a teszteleshez irhatsz nyugodtan sajat dolgokat is (JUnit nelkul is), szoval legkozelebbe ez ne gatoljon.

A logikara visszaterve: ok, ezt nem vettem eszre, hogy az attack reseteli a cooldownt. Ez igy teljesen ok, akkor nem szoltam. Illetve szoltam, es epp ez a lenyeg egy PR-nal, hogy ezek a kerdeses reszek tisztazodjanak, egyaltalan nem biztos, hogy a reviewer lat jol dolgokat, "vedjetek" meg a kodotokat, ha Nektek van igazatok. Mint pl most.